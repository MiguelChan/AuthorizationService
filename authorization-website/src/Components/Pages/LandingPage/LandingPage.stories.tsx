import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import { 
  LandingPage,
} from './LandingPage';

export default {
  title: 'Components/Pages/LandingPage',
  component: LandingPage,
} as Meta;

const Template: Story = () => <LandingPage />;

export const Primary = Template.bind({});