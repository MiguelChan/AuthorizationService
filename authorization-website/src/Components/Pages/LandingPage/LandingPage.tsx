import { Container, Typography } from '@mui/material';
import React from 'react';
import { AppToolbar } from '../../Blocks';
import { SimpleAppTemplate } from '../../Templates';

/**
 * .
 */
export const LandingPage: React.FunctionComponent = () => {

  const getApplicationToolbar = (): React.ReactElement => (
    <AppToolbar />
  );

  const getApplicationContent = (): React.ReactElement => (
    <Container>
      <Typography variant='h6'>Welcome to the Auth-Hub</Typography>
      <Typography variant='body1'>We're currently working on adding more features to the Website.</Typography>
      <Typography variant='body1'>Be patient until we finish.</Typography>
    </Container>
  );

  return (
    <SimpleAppTemplate 
      applicationToolbar={getApplicationToolbar()}
      applicationContent={getApplicationContent()}
    />
  );
};