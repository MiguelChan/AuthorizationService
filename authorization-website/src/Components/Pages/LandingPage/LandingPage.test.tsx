import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { LandingPage } from './LandingPage';

describe('LandingPage', () => {

  it('Should display the LandingPage', () => {
    const renderResult = setupComponent();
    expect(renderResult.container).toMatchSnapshot();
  });

  const setupComponent = (): RenderResult => {
    return render(<LandingPage />);
  }

});