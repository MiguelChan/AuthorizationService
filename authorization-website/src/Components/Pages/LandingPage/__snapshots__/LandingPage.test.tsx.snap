// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Should display the LandingPage 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        Auth-Hub
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Login
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-qhkpqj-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3ze8r2-MuiPaper-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Welcome to the Auth-Hub
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          We're currently working on adding more features to the Website.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Be patient until we finish.
        </p>
      </div>
    </div>
  </div>
</div>
`;
